xml/ha_config_basics.xml

```diff
@@ -2263,16 +2264,31 @@ monitor_0     interval=5s timeout=20s</screen>
     </para>
    </note>
    <para>
-    Apart from detecting the minimal requirements, the &hasi; also allows
-    to monitor the current utilization via the
+    Apart from detecting the minimal requirements, you can also
+    monitor the current utilization via the
     <systemitem>VirtualDomain</systemitem> resource agent. It detects CPU
     and RAM use of the virtual machine. To use this feature, configure a
     resource of the following class, provider and type:
     <literal>ocf:heartbeat:VirtualDomain</literal>. The following instance
-    attributes are available: <varname>autoset_utilization_cpu</varname> and
-    <varname>autoset_utilization_hv_memory</varname>. Both default to
-    <literal>true</literal>. This updates the utilization values in the CIB
-    during each monitoring cycle.
+    attributes are available:
+   </para>
+   <itemizedlist>
+    <listitem>
+     <para>
+      <varname>autoset_utilization_cpu</varname>
+     </para>
+    </listitem>
+    <listitem>
+     <para>
+      <varname>autoset_utilization_hv_memory</varname> (for Xen) or
+      <varname>autoset_utilization_host_memory</varname> (for KVM)
+     </para>
+    </listitem>
+   </itemizedlist>
+   <para>
+    These attributes default to <literal>true</literal>. This updates the utilization
+    values in the CIB during each monitoring cycle. For more information, see
```

Comment:
> Thanks. Looks good to me. Only a nitpicking.

@gao-yan

In my view, upstream creates unnecessary confusion and complexity  [1]. 

If we keep aligning this current upstream situation, for VirtualDomain and NodeUtilization with KVM, from my understanding, our document need call attention for users to understand and set NodeUtilization::utilization_hv_memory=false and VirtualDomain::autoset_utilization_hv_memory=false. Otherwise, NodeUtilization will set hv_memory="-512" in the run time, I doubt pacemaker will play well with "placement_strategy" as users want. What do you think?

@tahliar 

I suggest to get this on-hold for some days to see if upstream would like to debate my suggestion over there [1].

[1] https://github.com/ClusterLabs/resource-agents/pull/1649#issuecomment-1288410072

Reply:
> In my view, upstream creates unnecessary confusion and complexity [1].

@zzhou1 You made a very good point there. Indeed unnecessary complexity had been unfortunately introduced into both of the RAs... The only thing is, if to remove any existing parameters from the RAs,  it'd need to be carefully thought through so that it wouldn't break users' existing setups...

> If we keep aligning this current upstream situation, for VirtualDomain and NodeUtilization with KVM, from my understanding, our document need call attention for users to understand and set NodeUtilization::utilization_hv_memory=false and VirtualDomain::autoset_utilization_hv_memory=false. Otherwise, NodeUtilization will set hv_memory="-512" in the run time, 

IIRC in my testing, NodeUtilization created hv_memory="0" for my node, where there was no xentool installed... But either way, it's less of a problem that NodeUtilization provides a metric without a positive value, but it'll be a real problem if VirtualDomain decides to consume the metric ...
 
> [1] [ClusterLabs/resource-agents#1649 (comment)](https://github.com/ClusterLabs/resource-agents/pull/1649#issuecomment-1288410072)

Yes, let's hear some opinions from there.

Reply:
> NodeUtilization created hv_memory="0" for my node

Aha, yes, Yan is right. My bad eye ball checking skips one code line.

@tahliar 

So far, I don't see the immediate upstream action to cleanup both NodeUtilization and VirtualDomain, though I really think the code can be smarter.

To face the current reality, to make users' life easier to use both RA, to distinguish these hv_memory and host_memory concepts easier, probably we could add some further points like:

1) In practice, let's use host_memory concept for KVM, hv_memory concept for XEN.

2) the current 15SP4 code turns on both hv_memory and host_memory for NodeUtilization and VirtualDomain RA, no matter XEN or KVM. That is unnecessary. Please read the advice 1).

3) For KVM, the reference configuration 

```
primitive p_nu NodeUtilization \
        params utilization_hv_memory=false \
        op monitor timeout=20s interval=60

primitive p_vm VirtualDomain \
        params autoset_utilization_hv_memory=false \
        op monitor timeout=30s interval=10s
```

4) For XEN, the reference configuration 

```
primitive p_nu NodeUtilization \
        params utilization_host_memory=false \
        op monitor timeout=20s interval=60

primitive p_vm VirtualDomain \
        params autoset_utilization_host_memory=false \
        op monitor timeout=30s interval=10s
```

5) BTW, there are some typos in our DOC heritage, which was cooked for XEN many years ago. We could change this time

    From: "utilization memory"
    To: "utilization hv_memory"

Reply:
Thanks @zzhou1, I've added a couple of new commits: one to fix the typo, and one to add a note about the attributes (I'll also add a screenshot of the rendered version in another comment).